services/ui-src/src/components/report/Sidebar.test.tsx

```diff
@@ -18,60 +39,72 @@ const report = {
   ],
 };
 
-const mockReportStore = jest.fn().mockImplementation(() => ({
-  pageMap: mockPageMap,
-  report: report,
-  currentPageId: "id-1",
-  setCurrentPageId,
-}));
-jest.mock("../../utils/state/useStore", () => ({
-  useStore: () => mockReportStore(),
-}));
-
 describe("Sidebar", () => {
+  beforeEach(() => {
+    (useStore as unknown as jest.Mock).mockReturnValue({
+      pageMap: mockPageMap,
+      report,
+      currentPageId: "id-1",
+      setCurrentPageId,
+    });
+    (useParams as jest.Mock).mockReturnValue({
+      reportType: "exampleReport",
+      state: "exampleState",
+      reportId: "123",
+    });
+  });
   test("should not render if missing details from the store", () => {
-    mockReportStore.mockReturnValueOnce({
+    (useStore as unknown as jest.Mock).mockReturnValueOnce({
       pageMap: undefined,
       report: undefined,
       currentPageId: undefined,
       setCurrentPageId,
     });
-    const { container } = render(<Sidebar />);
-    expect(container).toBeEmptyDOMElement();
-  });
 
-  test("should not render if missing root in page map", () => {
-    mockReportStore.mockReturnValueOnce({
-      pageMap: new Map(),
-      report: report,
-      currentPageId: "id-1",
-      setCurrentPageId,
-    });
-    const { container } = render(<Sidebar />);
+    const { container } = render(
+      <Router>
+        <Sidebar />
+      </Router>
+    );
     expect(container).toBeEmptyDOMElement();
   });
 
   test("should render section headers", () => {
-    const { getByText } = render(<Sidebar />);
-    expect(getByText("Section 1")).toBeTruthy();
-    expect(getByText("Section 2")).toBeTruthy();
+    render(
+      <Router>
+        <Sidebar />
+      </Router>
+    );
+    expect(screen.getByText("Section 1")).toBeInTheDocument();
+    expect(screen.getByText("Section 2")).toBeInTheDocument();
   });
 
   test("should attempt to navigate on Click", async () => {
-    const { getByText } = render(<Sidebar />);
-    const button = getByText("Section 1");
+    render(
+      <Router>
+        <Sidebar />
+      </Router>
+    );
+    const button = screen.getByText("Section 1");
     await userEvent.click(button);
-    expect(setCurrentPageId).toHaveBeenCalledWith("id-1");
+    const reportPath = "/report/exampleReport/exampleState/123/id-1";
+    expect(mockUseNavigate).toHaveBeenCalledWith(reportPath);
   });
 
   test("should expand on Click", async () => {
-    const { getByText, queryByText } = render(<Sidebar />);
+    render(
+      <Router>
+        <Sidebar />
+      </Router>
+    );
+
+    const expandButton = screen.getByAltText("Expand subitems");
+    expect(screen.queryByText("Child 1")).not.toBeInTheDocument();
 
-    expect(getByText("Section 1")).toBeTruthy();
     await act(async () => {
-      const button = screen.getByAltText("Expand subitems");
-      await userEvent.click(button);
+      await userEvent.click(expandButton);
     });
```

Comment:
I'm fairly sure that this `act()` is unneeded. UserEvent is smart enough to perform any needed `act` calls internally.

Reply:
hmm... weird it fails without it 😬

Reply:
never mind, upon updating per previous comment, it works!